perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -1729,6 +1729,8 @@ private void mailValidation(Application app, ApplicationMail mail, List<Applicat
 									newValue += "?vo="+ getUrlEncodedString(app.getVo().getShortName());
 									newValue += ((app.getGroup() != null) ? "&group="+ getUrlEncodedString(app.getGroup().getName()) : EMPTY_STRING);
 									newValue += "&i=" + URLEncoder.encode(i, StandardCharsets.UTF_8) + "&m=" + URLEncoder.encode(m, StandardCharsets.UTF_8);
+									String redirectURL = BeansUtils.stringToMapOfAttributes(app.getFedInfo()).get("redirectURL");
+									newValue += (redirectURL != null) ? "&target=" + redirectURL : EMPTY_STRING;
```

Comment:
The notification template can contain either `{validationLink-[krb|non|...|]}` or `{validationLink}`. This is handling only the first case. For the second case, please add this as well. It is a couple of lines below.